api/v1alpha3/azuremachine_webhook.go

```diff
@@ -76,11 +76,15 @@ func (m *AzureMachine) ValidateUpdate(oldRaw runtime.Object) error {
 	machinelog.Info("validate update", "name", m.Name)
 	var allErrs field.ErrorList
 
-	if errs := ValidateOSDisk(m.Spec.OSDisk, field.NewPath("osDisk")); len(errs) > 0 {
+	old := oldRaw.(*AzureMachine)
```

Comment:
nit: Can we move this statement closer to it's usage. Looks like it's not used until line 101.